h/security/policy/combined.py

```diff
@@ -87,7 +87,4 @@ def _call_sub_policies(self, method, request, *args, **kwargs):
 
     @staticmethod
     def _is_api_request(request):
-        return request.path.startswith("/api") and request.path not in [
-            "/api/token",
-            "/api/badge",
-        ]
+        return request.matched_route and request.matched_route.name.startswith("api.")
```

Comment:
If you look closely at [`routes.py`](https://github.com/hypothesis/h/blob/ff26dc702f407a37a433e9c49a691d1fa30bc2d4/h/routes.py#L95-L192) all the API endpoints have route names starting with `"api."`. There are a couple of non-API endpoints whose paths nonetheless start with `"/api/"`: the token and badge endpoints. But the _route names_ for these endpoints do not start with `"api."` (they're just named `"token"` and `"badge"`). So `request.matched_route.name` is perfect for deciding whether a request is an API request.

This will also not return true for paths that start with `"/api/"` but don't exist like `"/api/does/not/exist"`.